galaxy/api/internal/views/search.py

```diff
@@ -0,0 +1,137 @@
+# (c) 2012-2019, Ansible by Red Hat
+#
+# This file is part of Ansible Galaxy
+#
+# Ansible Galaxy is free software: you can redistribute it and/or modify
+# it under the terms of the Apache License as published by
+# the Apache Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# Ansible Galaxy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# Apache License for more details.
+#
+# You should have received a copy of the Apache License
+
+from rest_framework.response import Response
+from rest_framework.settings import api_settings
+
+from galaxy.api import base
+from galaxy.api import exceptions
+from galaxy.api import serializers as serializers_v1
+from galaxy.api.internal import serializers as serializers_int
+from galaxy.api.internal.search import CollectionSearch, ContentSearch
+from galaxy.main import models
+
+
+__all__ = (
+    'SearchView',
+)
+
+CONTENT_ONLY_FILTERS = [
+    'platforms',
+    'clouds',
+]
+
+
+def _ensure_positive_int(string, field, cutoff=None):
+    msg = f'{field} must be a positive integer'
+    try:
+        value = int(string)
+    except ValueError:
+        raise exceptions.ValidationError(detail=msg)
+    if value < 0:
+        raise exceptions.ValidationError(detail=msg)
+    if cutoff is not None:
+        return min(value, cutoff)
+    return value
```

Comment:
if this caps the max value, it should probably cap the min value, too, instead of erroring in that case.

Reply:
IMO while user can easily specify max value bigger than our internal limit, negative value is definitely invalid.